pkg/sdk/action/download.go

```diff
@@ -89,13 +89,7 @@ func (action *DownloadAction) Download(fileInfo *storage.ResourceInfo) ([]byte,
 
 func (action *DownloadAction) initiateDownload(fileInfo *storage.ResourceInfo) (*gateway.InitiateFileDownloadResponse, error) {
 	// Initiating a download request gets us the download endpoint for the specified resource
-	req := &provider.InitiateFileDownloadRequest{
-		Ref: &provider.Reference{
-			Spec: &provider.Reference_Path{
-				Path: fileInfo.Path,
-			},
-		},
-	}
+	req := &provider.InitiateFileDownloadRequest{Ref: &provider.Reference{ResourceId: fileInfo.Id}}
```

Comment:
Let's leave this as `Path: fileInfo.Path`?

Reply:
`initiateDownload` uses a `ResourceInfo`. The only stable identifier we get from that is the id. Maybe it should use a new `Reference` as the parameter? 
```
func (action *DownloadAction) Download(fileInfo *storage.ResourceInfo) ([]byte, error) {
```
also should use a Reference?

Reply:
I can't really say, haven't looked at this code